src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -434,7 +385,10 @@ public SAMRecordIterator assertSorted(final SortOrder sortOrder) {
             final Container container = ContainerIO.readContainerHeader(newIterator.getCramHeader().getVersion().major, seekableStream);
             seekableStream.seek(seekableStream.position() + container.containerByteSize);
             iterator = newIterator;
-            iterator.jumpWithinContainerToPos(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, SAMRecord.NO_ALIGNMENT_START);
+            boolean atAlignments;
+            do {
+                atAlignments = iterator.jumpToAlignment(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, SAMRecord.NO_ALIGNMENT_START);
+            } while (!atAlignments && iterator.hasNext());
```

Comment:
What happens if the iterator runs out of containers before finding the requested position? Make sure this case is tested (eg., cram with no unmapped reads)